### PR TITLE
Add the standard preprocessor symbols for each target framework

### LINKED ISSUE #14: Add the framework-specific preprocessor flags automatically
MSBuild defines flags like `NETFRAMEWORK` and `NET48` automatically. We should match this behaviour.

We already support `defines` as of #1 . We should have a dict from tfm to a set of defines to automatically inject, and do this inside `AssemblyAction` based on `target_framework`.

### PR DESCRIPTION
**Pull request by j3parker**
Fixes #14

### CODE REVIEW

File: csharp/private/actions/assembly.bzl [hunk 1]
```diff
@@ -87,7 +103,8 @@ def AssemblyAction(
     args.add_all(resources, map_each = _format_resource_arg)
 
     # defines
-    args.add_joined("/define", defines, join_with=";")
+    _add_framework_preprocessor_symbols(args, target_framework)
+    args.add_joined("/define", defines, join_with=":")
```
Comment (j3parker) [inline csharp/private/actions/assembly.bzl]: Fixed a related typo here

Comment (omsmith) [reply csharp/private/actions/assembly.bzl]: You sure?

Comment (j3parker) [reply csharp/private/actions/assembly.bzl]: Yeah `;` vs `:`

Comment (j3parker) [reply csharp/private/actions/assembly.bzl]: Ohhhh

Comment (omsmith) [reply csharp/private/actions/assembly.bzl]: This is the separator for the different symbols

Comment (j3parker) [reply csharp/private/actions/assembly.bzl]: Ok I think the old thing would do `/define A;B` but what we want is either `/define:A;B` or `/define:A /define:B`. I'll back this out and we can fix it later.

Comment (j3parker) [reply csharp/private/actions/assembly.bzl]: I did the sad thing :(

**Comment by j3parker** [comment]
```
action 'Compiling lib_test'
  Mnemonic: CSharpCompile
  Target: //examples:lib_test
  Configuration: darwin-fastbuild
  ActionKey: 55b77863b25f449989a6f658d14a1ec6
  Inputs: [bazel-out/darwin-fastbuild/bin/examples/lib.ref.dll, examples/libtest.cs, external/NUnit/lib/net45/nunit.framework.dll, external/NUnitLite/lib/net45/nunitlite.dll, external/csharp-build-tools/tasks/netcoreapp2.1/bincore/csc.dll, external/d2l_rules_csharp/csharp/private/nunit/shim.cs, external/net472/build/.NETFramework/v4.7.2/Facades/System.Linq.dll, external/net472/build/.NETFramework/v4.7.2/System.Core.dll, external/net472/build/.NETFramework/v4.7.2/mscorlib.dll, external/netcore-runtime-osx/dotnet]
  Outputs: [bazel-out/darwin-fastbuild/bin/examples/lib_test.exe, bazel-out/darwin-fastbuild/bin/examples/lib_test.pdb, bazel-out/darwin-fastbuild/bin/examples/lib_test.ref.exe]
  Command Line: (exec external/netcore-runtime-osx/dotnet \
    external/csharp-build-tools/tasks/netcoreapp2.1/bincore/csc.dll \
    /noconfig \
    /unsafe- \
    /checked- \
    /nostdlib+ \
    /utf8output \
    /deterministic+ \
    /filealign:512 \
    /nologo \
    /highentropyva \
    /warn:0 \
    /target:exe \
    /langversion:7.3 \
    /debug+ \
    /optimize- \
    /out:bazel-out/darwin-fastbuild/bin/examples/lib_test.exe \
    /refout:bazel-out/darwin-fastbuild/bin/examples/lib_test.ref.exe \
    /pdb:bazel-out/darwin-fastbuild/bin/examples/lib_test.pdb \
    /r:external/net472/build/.NETFramework/v4.7.2/mscorlib.dll \
    /r:external/net472/build/.NETFramework/v4.7.2/System.Core.dll \
    /r:external/net472/build/.NETFramework/v4.7.2/Facades/System.Linq.dll \
    /r:bazel-out/darwin-fastbuild/bin/examples/lib.ref.dll \
    /r:external/NUnitLite/lib/net45/nunitlite.dll \
    /r:external/NUnit/lib/net45/nunit.framework.dll \
    examples/libtest.cs \
    external/d2l_rules_csharp/csharp/private/nunit/shim.cs \
    /d:NETFRAMEWORK \
    /d:NET472)
```

File: csharp/private/actions/assembly.bzl (hunk 1 above)
Comment (omsmith) [reply csharp/private/actions/assembly.bzl]: Looking at the code before, it definitely wasn't obvious that it supported `/define` being specified multiple times

**Pull request merged**